proctor-common/src/main/java/com/indeed/proctor/common/model/TestType.java

```diff
@@ -34,6 +39,18 @@ public static TestType register(@Nonnull final String name) {
         return previous != null ? previous : testType;
     }
 
+    public void addDependency(@Nonnull final TestType dependency) {
```

Comment:
As similar to https://github.com/indeedeng/proctor/pull/74/files#r912645823, I think it's nice to use a longer name `addAllowedDependency`. This name sounds like one test type depends on another type.